pages/en/supported-networks/cosmos.mdx

```diff
@@ -0,0 +1,240 @@
+  validatorUpdates: Array<Validator>
+}
+
+class EventData {
+  event: Event
+  block: HeaderOnlyBlock
+}
+
+class TransactionData {
+  tx: TxResult
+  block: HeaderOnlyBlock
+}
+
+class HeaderOnlyBlock {
+  header: Header
+}
+
+class Header {
+  version: Consensus
+  chainId: string
+  height: u64
+  time: Timestamp
+  lastBlockId: BlockID
+  lastCommitHash: Bytes
+  dataHash: Bytes
+  validatorsHash: Bytes
+  nextValidatorsHash: Bytes
+  consensusHash: Bytes
+  appHash: Bytes
+  lastResultsHash: Bytes
+  evidenceHash: Bytes
+  proposerAddress: Bytes
+  hash: Bytes
+}
+
+class TxResult {
+  height: u64
+  index: u32
+  tx: Bytes
+  result: ResponseDeliverTx
+  hash: Bytes
+}
+
+class Event {
+  eventType: string
+  attributes: Array<EventAttribute>
+}
+```
+
+The types above are just the general ones that mappings use. You can find the full list of types for the Cosmos integration [here](https://github.com/graphprotocol/graph-ts/blob/4c064a8118dff43b110de22c7756e5d47fcbc8df/chain/cosmos.ts).
+
+Each type of handler will receive a different type based on the relevant data. For both event and transaction handlers, a reference to the block they are contained in is passed as well. These are the exact types that are passed as a parameter to each mapping function:
+
+`Block` is passed to the blockHandler.
+
+`EventData` is passed to the eventHandler.
+
+`TransactionData` is passed to the transactionHandler.
+
+## Creating and building a Cosmos subgraph
+
+The first step before starting to write the subgraph mappings is to generate the type bindings based on the entities that have been defined in the subgraph schema file (`schema.graphql`). This will allow the mapping functions to create new objects of those types and save them to the store. This is done by using the `codegen` CLI command:
+
+```bash
+$ graph codegen
+```
+
+Once the mappings are ready, the subgraph needs to be built. This step will highlight any errors the manifest or the mappings might have. A subgraph needs to build successfully in order to be deployed to the Graph Node. It can be done using the `build` CLI command:
+
+```bash
+$ graph build
+```
+
+## Deploying a Cosmos subgraph
+
+When the subgraph builds successfully, it is time to deploy it to a Graph Node instance for indexing. Cosmos subgraphs can be deployed to any Graph Node on version `>=v0.26.0`.
+
+The first thing to do is create the subgraph in the node, this only has to be done once via the `create` CLI command:
+
+```bash
+Local deployment:
+$ graph create subgraph-name --node http://localhost:8020
+
+Hosted service deployment:
+$ graph create subgraph-name --product hosted-service
+```
+
+After the subgraph has been created, it can be deployed by using the graph `deploy` CLI command:
+
+```bash
+Local deployment:
+$ graph deploy subgraph-name --ipfs http://localhost:5001 --node http://localhost:8020
+
+Hosted service deployment:
+$ graph deploy subgraph-name --product hosted-service
+```
```

Comment:
Once your subgraph has been created, you can deploy your subgraph by using the `graph deploy` CLI command after running the `graph create` CLI command:
#### Hosted Service:
```bash
graph create subgraph-name --product hosted-service # creates a subgraph on a local Graph Node (on the Hosted Service, this is done via the UI)
```
```bash
graph deploy --node https://api.thegraph.com/deploy/ --ipfs https://api.thegraph.com/ipfs/ --access-token <your-access-token>
```

#### Local Graph Node (based on default configuration):
```bash
graph create subgraph-name --node http://localhost:8020
```
```bash
graph deploy --node http://localhost:8020/ --ipfs http://localhost:5001
```